data_dump/README.md

```diff
@@ -0,0 +1,152 @@
+dataset contains the dimensions `report_step` as dates and `index` as the
+`GEN_DATA` indexing. The dates have been obtained by indexing the `time-map`
+vector at the given `REPORT_STEP` indices. Each NetCDF variable is a separate
+`GEN_DATA` entry.
+
+## `SUMMARY`
+
+All `SUMMARY`s are in the same group under `/REAL_{index}/SUMMARY`. This dataset
+contains the dimension `report_step`, which is the `time-map`. Each NetCDF
+variable is a separate `SUMMARY` entry.
+
+## `FIELD`
+
+Each ERT `FIELD` is a subgroup of `/REAL_{index}/FIELD`, where the name of the
+group is the name of the field. The dimensions are `x`, `y` and `z`, which
+correspond to the 3-D axes of fields. The data for the field is given by the
+`VALUES` variable.
+
+## `SURFACE`
+
+Each ERT `SURFACE` is a subgroup of `/REAL_{index}/SURFACE`, where the name of
+the group is the name of the surface. The data for the surface is given by the
+`VALUES` variable.
+
+# Reading
+
+To programmatically read data from the files it is recommended to use both the
+`netCDF4` package for low-level access, and `xarray` package for manipulating or
+converting the data.
+
+The `xarray` package does not make it easy to navigate groups which is why
+`netCDF4` is used. `xarray` uses `netCDF4` as a backend to open NetCDF version 4
+files, so both packages need to be installed anyway.
+
+## `netCDF4` package
+
+Groups can be navigated using `netCDF4`. First, open a file:
+
+``` python
+import netCDF4
+
+ds = netCDF4.Dataset("some-dataset.nc")
+```
+
+Groups are given using the `.groups` property. This returns a Python `dict` with
+the keys being the name of the group, and the value being a `netCDF4.Group`
+object. Groups may have subgroups.
+
+``` python
+for name, group in ds.groups.items():
+    for subname, subgroup in group.group.items():
+        print(f"{name}/{subname} = {subgroup}")
+```
+
+Specific subgroups can be accessed by `ds.groups["some_group"]` or
+`ds.group["some_group"]`. It is also possible to access subgroups by path, as if
+in a file structure: `ds["/foo/bar/baz"]`, which is equivalent to
+`ds.groups["foo"].groups["bar"].groups["baz"]`.
+
+NetCDF datasets have _variables_, which store actual data, and _dimensions_ (aka
+_coordinates_), which are the axes for the data. Within a group, if two
+different variables refer to a dimension by one name, then both variables will
+have the same dimension values. For example, the variables for porosity and
+pressure are different, but may share the same `x`, `y`, `z` axes.
+
+To access dimensions, use the `.dimensions` property. To access variables, use
+the `.variables` property. Both return Python `dict`s with keys as the dimension
+or variable names and the values are `netCDF4` data class instances. Note that
+dimensions are also variables. Unlike `xarray`, the low-level `netCDF4` package
+has no concept of "data variable". To get the data variables, do:
+
+``` python
+data_vars = {
+  key: val
+  for key, val in ds.variables.items()
```

Comment:
`ds` here has to refer to a Group I think, because otherwise we can't get `variables`.
Perhaps write `group.variables.items()` instead?

Reply:
Or hm, I may be misunderstanding. I am getting that `ds.variables` is an empty dict, but perhaps that's as expected and OK..

Reply:
Groups are datasets with a group name, basically. The outermost dataset is a group with the name `/` ("root group"). So they're kinda the same thing.

So, if you save a dataset to group `/REAL_0/GEN_KW/COEFFS`, then there are 4 datasets in the file, of which 3 are empty (`/`, `/REAL_0`, `/REAL_0/GEN_KW`) and 1 with variables (`/REAL_0/GEN_KW/COEEFS`).